motoman_driver/include/motoman_driver/simple_message/messages/motoman_read_single_io_message.h

```diff
@@ -0,0 +1,126 @@
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * \author G.A. vd. Hoorn (TU Delft Robotics Institute)
+ */
+
+#ifndef MOTOMAN_DRIVER_SIMPLE_MESSAGE_MOTOMAN_READ_SINGLE_IO_MESSAGE_H
+#define MOTOMAN_DRIVER_SIMPLE_MESSAGE_MOTOMAN_READ_SINGLE_IO_MESSAGE_H
+
+#ifdef ROS
+#include "simple_message/typed_message.h"
+#include "simple_message/shared_types.h"
+#include "motoman_driver/simple_message/motoman_simple_message.h"
+#include "motoman_driver/simple_message/motoman_read_single_io.h"
+
+#endif
+
+#ifdef MOTOPLUS
+#include "typed_message.h"
+#include "shared_types.h"
+#include "motoman_simple_message.h"
+#include "motoman_read_single_io.h"
+
+#endif
+
+namespace motoman
+{
+namespace simple_message
+{
+namespace io_ctrl_message
+{
+/**
+ * \brief Class encapsulated motoman read single io message generation methods
+ * (either to or from a industrial::simple_message::SimpleMessage type.
+ *
+ * This message simply wraps the following data type:
+ *   motoman::simple_message::io_ctrl::ReadSingleIO
+ * The data portion of this typed message matches ReadSingleIO exactly.
+ *
+ *
+ * THIS CLASS IS NOT THREAD-SAFE
+ *
+ */
+class ReadSingleIOMessage : public industrial::typed_message::TypedMessage
+{
+public:
+  /**
+   * \brief Default constructor
+   *
+   * This method creates an empty message.
+   *
+   */
+  ReadSingleIOMessage(void);
+  /**
+   * \brief Destructor
+   *
+   */
+  ~ReadSingleIOMessage(void);
+  /**
+   * \brief Initializes message from a simple message
+   *
+   * \param simple message to construct from
+   *
+   * \return true if message successfully initialized, otherwise false
+   */
+  bool init(industrial::simple_message::SimpleMessage & msg);
+
+  /**
+   * \brief Initializes message from a read single io structure
+   *
+   * \param cmd read single io data structure
+   *
+   */
+  void init(motoman::simple_message::io_ctrl::ReadSingleIO & cmd);
+
+  /**
+   * \brief Initializes a new message
+   *
+   */
+  void init();
+
+  // Overrides - SimpleSerialize
+  bool load(industrial::byte_array::ByteArray *buffer);
+  bool unload(industrial::byte_array::ByteArray *buffer);
+
+  unsigned int byteLength()
+  {
+    return this->cmd_.byteLength();
+  }
+
+  motoman::simple_message::io_ctrl::ReadSingleIO cmd_;

```

Comment:
Yes, that might make sense.

None of the other message classes have the wrapped data structure as a private member though, so if we want to maintain consistency we should probably leave it public.

I would welcome a PR that makes all wrapped structures private though.